main.go

```diff
@@ -16,6 +17,9 @@ func main() {
 	if err := cmd.Run(context.Background(), os.Args[1:]...); err != nil {
 		var userErr cmd.Error
 		switch {
+		case strings.HasSuffix(err.Error(), ": EOF"):
+			logging.Debugf("%v\n", err)
+			fmt.Fprintln(os.Stderr, "Could not reach infra server, please wait a moment and try again.")
```

Comment:
A couple thoughts here:

Hopefully we can use `errors.Is(err, io.EOF)` here to detect the error. Checking for the `: EOF` suffix should be relatively safe as well, but if `errors.Is` does not work for some reason it might mean we are wrapping errors incorrectly, which would be a good thing to fix.

I think handling the error at this point (in the `main` function) risks that the EOF could come from other places than the http request to the server. It could have come from reading a file or other network operation.

I think this check would be safer if we moved it to `attemptTLSRequest`. I think the bottom of that function is where we generally see it, and we already have some error handling there.

If we want to apply it to other requests as well, we handle the request errors in `api/client.go:request`. 

Both of these functions have similar checks for `errors.As(err, urlErr)` to check for a timeout. Maybe we should refactor that out into a function something like this, and call it from the two places?

```go
func handleConnError(err error) error {
	if err == nil {
		return nil
	}
	urlErr := &url.Error{}
	if errors.As(err, &urlErr) {
		if urlErr.Timeout() {
			return nil, fmt.Errorf("%w: %s", ErrTimeout, err)
		}
	}
	if errors.Is(err, io.EOF) {
		return fmt.Errorf("could not reach infra server, please wait a moment and try again")
	}
	return err
}
```